lobot/js/contentGrid.js

```diff
@@ -0,0 +1,54 @@
+(async function ContentGrid() {
+  const contentCards = document.querySelectorAll(`.content-grid .card`);
+  contentCards.forEach((element) => {
+    element.addEventListener("click", async function (event) {
+      event.preventDefault();
+      const urlFragments = event.currentTarget.closest("a").href.split("/");
+      const contentId = urlFragments[4];
+      const requestPath = `${urlFragments[4]}`;
+      const response = fetch(requestPath, {
+        method: "GET",
+        credentials: "same-origin",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }).then((response) => {
+        response.json().then((body) => {
+          const content = JSON.parse(body);
+          const contentGrid = document.querySelector(`.content-grid`);
+          const contentContainer = document.querySelector(`.content-container`);
```

Comment:
Refactor to outer closure so we can use this in multiple functions to remove duplication.